File: app/src/main/java/com/viginfotech/chennaitimes/firebase/SyncSheduleService.java
package com.viginfotech.chennaitimes.firebase;

import android.content.Intent;
import android.util.Log;

import com.firebase.jobdispatcher.JobParameters;
import com.firebase.jobdispatcher.JobService;
import com.viginfotech.chennaiTimesApi.ChennaiTimesApi;
import com.viginfotech.chennaitimes.Config;
import com.viginfotech.chennaitimes.data.NewsContract;
import com.viginfotech.chennaitimes.sync.TriggerRefresh;
import com.viginfotech.chennaitimes.util.CloudEndpointBuilderHelper;
import com.viginfotech.chennaitimes.util.NetworkUtil;




/**
 * Created by anand on 6/11/17.
 */

public class SyncSheduleService extends JobService {

    String TAG="SyncScheduleService";
    private ChennaiTimesApi myApiService;

    @Override
    public void onCreate() {
        super.onCreate();
        if (myApiService == null) {
            myApiService = CloudEndpointBuilderHelper.getEndpoints();
        }
    }

    @Override
    public boolean onStartJob(JobParameters jobParameters) {
        Log.i(TAG, "onStartJob: ");
        if(jobParameters.getTag().equals(Config.SYNCSCHEDULE_TAG))
        syncAll();
        else if(jobParameters.getTag().equals(Config.TRUNCATE_TAG)){
            getContentResolver().delete(NewsContract.NewsEntry.CONTENT_URI, null,null);
        }
        return false;
    }

    private void syncAll() {
        if(!NetworkUtil.isOnline(this)) return;
       startService(new Intent(this, TriggerRefresh.class).putExtra("category",0).setAction("autosync"));
       startService(new Intent(this, TriggerRefresh.class).putExtra("category",1));
       startService(new Intent(this, TriggerRefresh.class).putExtra("category",2));
       startService(new Intent(this, TriggerRefresh.class).putExtra("category",3));
       startService(new Intent(this, TriggerRefresh.class).putExtra("category",4));
       startService(new Intent(this, TriggerRefresh.class).putExtra("category",5));
       startService(new Intent(this, TriggerRefresh.class).putExtra("category",6));


    }



    @Override
    public boolean onStopJob(JobParameters jobParameters) {
        return false;
    }
}

File: app/src/main/java/com/viginfotech/chennaitimes/firebase/SyncSheduleService.kt
package com.viginfotech.chennaitimes.firebase

import android.content.Intent
import android.util.Log
import com.firebase.jobdispatcher.JobParameters
import com.firebase.jobdispatcher.JobService
import com.viginfotech.chennaitimes.Config
import com.viginfotech.chennaitimes.data.NewsContract
import com.viginfotech.chennaitimes.service.ChennaiTimeAPIBuilder
import com.viginfotech.chennaitimes.sync.TriggerRefresh


/**
 * Created by anand on 6/11/17.
 */

class SyncSheduleService : JobService() {

    internal var TAG = "SyncScheduleService"
     val myApiService = ChennaiTimeAPIBuilder.buildChennaiTimesService();

    override fun onCreate() {
        super.onCreate()

    }

    override fun onStartJob(jobParameters: JobParameters): Boolean {
        Log.i(TAG, "onStartJob: ")
        if (jobParameters.tag == Config.SYNCSCHEDULE_TAG)
            syncAll()
        else if (jobParameters.tag == Config.TRUNCATE_TAG) {
            contentResolver.delete(NewsContract.NewsEntry.CONTENT_URI, null, null)
        }
        return false
    }

    private fun syncAll() {
        startService(Intent(this, TriggerRefresh::class.java).putExtra("category", 0))
        startService(Intent(this, TriggerRefresh::class.java).putExtra("category", 1))
        startService(Intent(this, TriggerRefresh::class.java).putExtra("category", 2))
        startService(Intent(this, TriggerRefresh::class.java).putExtra("category", 3))
        startService(Intent(this, TriggerRefresh::class.java).putExtra("category", 4))
        startService(Intent(this, TriggerRefresh::class.java).putExtra("category", 5))
        startService(Intent(this, TriggerRefresh::class.java).putExtra("category", 6))

    }

    override fun onStopJob(jobParameters: JobParameters): Boolean {
        return false
    }
}
